### ISSUE TITLE
Investigate/fix decoder's ValueError and DecodeError

### ISSUE THREAD

**Issue by hkrishna** [mapzen/mapbox-vector-tile#8]
As per #7 

### ValueError

```
curl "http://cdn.maplytic.no/kartverket.maplytic.no/tile/sosi_n50,sosi_n250,sosi_n1000,sosi_n2000,sosi_n5000/13/4245/2423.vector.pbf?sosi_n250:zoom=10-11&sosi_n5000:zoom=0-4&sosi_n50:zoom=12-20&sosi_n2000:zoom=5-6&featuretype=%5BOBJTYPE%5D&sosi_n1000:zoom=7-9" >> tile.pbf
```

```
>>> import mapbox_vector_tile
>>> data = open('tile.pbf', 'rb').read()
>>> print mapbox_vector_tile.decode(data)
```

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mapbox_vector_tile/__init__.py", line 9, in decode
    message = vector_tile.getMessage(tile)
  File "mapbox_vector_tile/decoder.py", line 39, in getMessage
    geometry = self.parse_geometry(feature.geometry, feature.type)
  File "mapbox_vector_tile/decoder.py", line 67, in parse_geometry
    cmd  = int(item[(ilen-cmd_bits):ilen], 2)
ValueError: invalid literal for int() with base 2: 'b10'
```

### DecodeError

```
curl "https://vector.mapzen.com/osm/all/16/19297/24639.mapbox" >> tile2.pbf
```

```
>>> import mapbox_vector_tile
>>> data2 = open('tile2.pbf', 'rb').read()
>>> print mapbox_vector_tile.decode(data2)
```

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mapbox_vector_tile/__init__.py", line 9, in decode
    message = vector_tile.getMessage(tile)
  File "mapbox_vector_tile/decoder.py", line 22, in getMessage
    self.tile.ParseFromString(pbf_data)
  File "/usr/local/lib/python2.7/site-packages/protobuf-2.6.1-py2.7.egg/google/protobuf/message.py", line 186, in ParseFromString
  File "/usr/local/lib/python2.7/site-packages/protobuf-2.6.1-py2.7.egg/google/protobuf/internal/python_message.py", line 841, in MergeFromString
  File "/usr/local/lib/python2.7/site-packages/protobuf-2.6.1-py2.7.egg/google/protobuf/internal/python_message.py", line 866, in InternalParse
  File "/usr/local/lib/python2.7/site-packages/protobuf-2.6.1-py2.7.egg/google/protobuf/internal/decoder.py", line 827, in SkipField
  File "/usr/local/lib/python2.7/site-packages/protobuf-2.6.1-py2.7.egg/google/protobuf/internal/decoder.py", line 797, in _RaiseInvalidWireType
google.protobuf.message.DecodeError: Tag had invalid wire type.
```

**Comment by relet** [comment]
FYI - this is what I decode using the library from https://github.com/ElectronicChartCentre/java-vector-tile

````
./validate.sh tile.pbf |head
LINESTRING (12.5 256, 12.5 255.875, 12.5 250.25)
{gid=749126, OBJTYPE=FiktivDelelinje, OPPDATERINGSDATO=10000101}
LINESTRING (12.5 89.3125, 12.5 86.25, 12.5 79.0625, 12.5 50.9375, 12.5 32.6875, 12.5 22.8125, 12.5 16.25, 12.5 3.3125, 12.5 0)
{gid=750434, OBJTYPE=FiktivDelelinje, OPPDATERINGSDATO=10000101}
LINESTRING (12.5 203, 18.5 203)
{gid=751427, OBJTYPE=FiktivDelelinje, OPPDATERINGSDATO=10000101}
LINESTRING (127.5 203, 135.9375 203, 150 203, 157.4375 203, 166.6875 203, 170.6875 203, 172.875 203, 180.5 203, 187.75 203, 189.125 203)
{gid=751432, OBJTYPE=FiktivDelelinje, OPPDATERINGSDATO=10000101}
LINESTRING (12.5 130.1875, 12.5 125.125, 12.5 104.4375, 12.5 96.625, 12.5 93.6875)
{gid=753175, OBJTYPE=FiktivDelelinje, OPPDATERINGSDATO=10000101}
````

````
./validate.sh tile2.pbf 
com.google.protobuf.InvalidProtocolBufferException: Protocol message tag had invalid wire type.
````

**Comment by hkrishna** [comment]
closing this thanks to @kaidamasaki fix - #14
